layers/dhcpv6.go

```diff
@@ -0,0 +1,495 @@
+		return "ReconfigureMessage"
+	case DHCPv6OptReconfigureAccept:
+		return "ReconfigureAccept"
+	default:
+		return "Unknown"
+	}
+}
+
+// DHCPv6Options is used to get nicely printed option lists which would normally
+// be cut off after 5 options.
+type DHCPv6Options []DHCPv6Option
+
+// String returns a string version of the options list.
+func (o DHCPv6Options) String() string {
+	buf := &bytes.Buffer{}
+	buf.WriteByte('[')
+	for i, opt := range o {
+		buf.WriteString(opt.String())
+		if i+1 != len(o) {
+			buf.WriteString(", ")
+		}
+	}
+	buf.WriteByte(']')
+	return buf.String()
+}
+
+// DHCPv6Option rerpresents a DHCP option.
+type DHCPv6Option struct {
+	Code   DHCPv6Opt
+	Length uint16
+	Data   []byte
+}
+
+// String returns a string version of a DHCP Option.
+func (o DHCPv6Option) String() string {
+	switch o.Code {
+	case DHCPv6OptClientID, DHCPv6OptServerID:
+		duid, err := decodeDuid(o.Data)
+		if err != nil {
+			return fmt.Sprintf("Option(%s:INVALID)", o.Code)
+		}
+		return fmt.Sprintf("Option(%s:[%s])", o.Code, duid.String())
+	case DHCPv6OptOro:
+		options := ""
+		for i := 0; i < int(o.Length); i += 2 {
+			if options != "" {
+				options += ","
+			}
+			option := DHCPv6Opt(binary.BigEndian.Uint16(o.Data[i : i+2]))
+			options += option.String()
+		}
+		return fmt.Sprintf("Option(%s:[%s])", o.Code, options)
+	default:
+		return fmt.Sprintf("Option(%s:%v)", o.Code, o.Data)
+	}
+}
+
+// NewDHCPv6Option constructs a new DHCPv6Option with a given type and data.
+func NewDHCPv6Option(code DHCPv6Opt, data []byte) DHCPv6Option {
+	o := DHCPv6Option{Code: code}
+	if data != nil {
+		o.Data = data
+		o.Length = uint16(len(data))
+	}
+
+	return o
+}
+
+func (o *DHCPv6Option) encode(b []byte) error {
+	binary.BigEndian.PutUint16(b[0:2], uint16(o.Code))
+	binary.BigEndian.PutUint16(b[2:4], o.Length)
+	copy(b[4:], o.Data)
+
+	return nil
+}
+
+func (o *DHCPv6Option) decode(data []byte) error {
+	if len(data) < 2 {
+		return errors.New("Not enough data to decode")
+	}
+	o.Code = DHCPv6Opt(binary.BigEndian.Uint16(data[0:2]))
+	if len(data) < 3 {
+		return errors.New("Not enough data to decode")
+	}
+	o.Length = binary.BigEndian.Uint16(data[2:4])
+	o.Data = data[4 : 4+o.Length]
+	return nil
+}
+
+// DHCPv6DuidType represents a DHCP DUID - RFC-3315
+type DHCPv6DuidType uint16
+
+// Constants for the DHCPv6DuidType.
+const (
+	DHCPv6DuidTypeLLT DHCPv6DuidType = iota + 1
+	DHCPv6DuidTypeEN
+	DHCPv6DuidTypeLL
+)
+
+// String returns a string version of a DHCPv6DuidType.
+func (o DHCPv6DuidType) String() string {
+	switch o {
+	case DHCPv6DuidTypeLLT:
+		return "LLT"
+	case DHCPv6DuidTypeEN:
+		return "EN"
+	case DHCPv6DuidTypeLL:
+		return "LL"
+	default:
+		return "Unknown"
+	}
+}
+
+// Duid means DHCP Unique Identifier as stated in RFC 3315, section 9 (https://tools.ietf.org/html/rfc3315#page-19)
+type Duid struct {
+	Type DHCPv6DuidType
+	// LLT, LL
+	HardwareType []byte
+	// EN
+	EnterpriseNumber []byte
+	// LLT
+	Time []byte
```

Comment:
I don't think so. In https://tools.ietf.org/html/rfc3315#page-19 it mentions `The time value is the time that the DUID is generated represented in seconds since midnight (UTC), January 1, 2000, modulo 2^32`. I didn't find anything related in https://golang.org/pkg/time/#Time nor in other packages (I maybe wrong)

Reply:
Sounds good, let's keep this one as is.  Maybe a helper function to translate this into a time.Time would be useful, but isn't necessary.